src/hotspot/share/prims/jvmtiEventController.cpp

```diff
@@ -985,12 +985,14 @@ JvmtiEventControllerPrivate::change_field_watch(jvmtiEvent event_type, bool adde
   if (added) {
     (*count_addr)++;
     if (*count_addr == 1) {
+      JvmtiVTMSTransitionDisabler disabler;
```

Comment:
Could you just declare this outside the if-block rather than repeating it?

Reply:
Thank you for looking at this and for the suggestion, David.
I wanted to optimize a little bit but agree it is not worth it, so updated now.